client/my-sites/media/details.jsx

```diff
@@ -0,0 +1,247 @@
+		this.props.onBack();
+	};
+
+	setDetailSelectedIndex = index => {
+		page( '/media/' + this.props.site.slug + '/' + this.getSelectedMedia( index ).ID );
+	};
+
+	/**
+	 * Start the process to delete media items.
+	 * `callback` is an optional parameter which will execute once the confirm dialog is accepted.
+	 * It's used especially when the item is attempting to be removed using the item detail dialog.
+	 *
+	 * @param  {Function} [callback] - callback function
+	 */
+	deleteMedia( callback ) {
+		const { translate, items } = this.props;
+		const selectedCount = items.length;
+		const confirmMessage = translate(
+			'Are you sure you want to delete this item? ' +
+				'Deleted media will no longer appear anywhere on your website, including all posts, pages, and widgets. ' +
+				'This cannot be undone.',
+			'Are you sure you want to delete these items? ' +
+				'Deleted media will no longer appear anywhere on your website, including all posts, pages, and widgets. ' +
+				'This cannot be undone.',
+			{ count: selectedCount }
+		);
+
+		accept(
+			confirmMessage,
+			accepted => {
+				if ( ! accepted ) {
+					return;
+				}
+
+				this.confirmDeleteMedia();
+				if ( callback ) {
+					callback();
+				}
+			},
+			translate( 'Delete' ),
+			null,
+			{
+				isScary: true,
+			}
+		);
+	}
+
+	enableEditMode = () => {
+		page( '/media/edit/' + this.props.site.slug + '/' + this.getSelectedMedia().ID );
+		return;
+	};
+
+	handleDeleteMediaEvent = () => {
+		this.deleteMedia();
+	};
+
+	deleteMediaByItemDetail = () => {
+		this.deleteMedia( () => this.closeDetailsModal() );
+	};
+
+	confirmDeleteMedia = () => {
+		const site = this.props.site;
+
+		if ( ! site ) {
+			return;
+		}
+
+		const selected =
+			this.props.items && this.props.items.length
+				? this.props.items
+				: MediaLibrarySelectedStore.getAll( site.ID );
+
+		MediaActions.delete( site.ID, selected );
+	};
+
+	getSelectedMedia = ( index = null ) => {
+		if ( index === null ) {
+			return this.props.items[ this.props.selected ];
+		}
+		return this.props.items[ index ];
+	};
+
+	render() {
+		const { site, view, items, selected } = this.props;
+		const mimePrefix = getMimePrefix( this.getSelectedMedia() );
+		return (
+			<Dialog
+				isVisible={ true }
+				additionalClassNames="editor-media-modal media__item-dialog"
+				buttons={ this.getModalButtons() }
+				onClose={ this.props.onBack }
+			>
+				{ view === 'single' && (
+					<EditorMediaModalDetail
+						site={ site }
+						backText="Back"
+						items={ items }
+						selectedIndex={ selected }
+						onReturnToList={ this.props.onBack }
+						onEditImageItem={ this.enableEditMode }
+						onEditVideoItem={ this.enableEditMode }
+						onRestoreItem={ this.restoreOriginalMedia }
+						onSelectedIndexChange={ this.setDetailSelectedIndex }
+					/>
+				) }
+				{ view === 'edit' &&
+					'video' !== mimePrefix && (
+						<ImageEditor
+							siteId={ site && site.ID }
```

Comment:
`siteId` is expected to be a number or `null`, but in this case it might end up being a boolean.